backend/src/controllers/viewed-cars/viewed-cars.controller.ts

```diff
@@ -0,0 +1,57 @@
+import * as viewedCarsService from '@services/viewed-cars/viewed-cars.service';
+import httpStatus from 'http-status-codes';
+
+import type {
+  setViewedCarRequest,
+  setViewedCarResponse,
+} from '@autoline/shared';
+import type { TypedRequestQuery } from '@common/types/controller/controller';
+import type { getViewedCarsListResponse } from '@common/types/types';
+import type { WishlistInput } from '@common/types/types';
+import type { NextFunction, Response } from 'express';
+
+const getList = async (
+  req: TypedRequestQuery<WishlistInput>,
+  res: Response<(getViewedCarsListResponse | null | undefined)[]>,
+  next: NextFunction,
+): Promise<void> => {
+  try {
+    const { userId } = req.params;
+    const result = await viewedCarsService.getList(userId);
+    res.json(result).status(httpStatus.OK);
+  } catch (error) {
+    console.error(error);
+    if (error instanceof Error) {
+      res.sendStatus(400);
+    }
+    next(error);
+  }
+};
+
+const setCar = async (
+  req: TypedRequestQuery<setViewedCarRequest>,
+  res: Response<setViewedCarResponse>,
+  next: NextFunction,
+): Promise<void> => {
+  try {
+    const { userId } = req.params;
+    const { modelId, complectationId } = req.query;
+
+    const reviewedCarsList = {
+      userId,
+      modelId,
+      complectationId,
+    };
```

Comment:
This object is misnamed. It's not the `List`

Reply:
done